### DISCUSSION TITLE
Third party database solutions

### DISCUSSION THREAD

**Discussion by naturzukunft** [eclipse-rdf4j/rdf4j#3363] [General]
Hi there, are there any Third party database solutions that can be used out of the box? without "minimal amount of code changes" ?
For example Halyard it sounds like it is usable with http repo ?

**Comment by naturzukunft** [comment]
oops,Halyard is already dead :-(

**Reply by abrokenjester** [reply]
Shame that - it looked very promising I thought.

**Comment by barthanssens** [comment]
Well, there is an experimental ElasticSearch storage sail included in RDF4J, maybe that could be helpful ?

There are also non-free (licensing fees to be paid) options like OntoText GraphDB which work very well with RDF4j (probably Stardog as well), and quite recently Oracle announced their RDF Graph Adapter for Eclipse RDF4J. Amazon Neptune DB wouth be another option.

**Reply by abrokenjester** [reply]
There's an overview here: https://rdf4j.org/about/#third-party-database-solutions (not claiming completeness of course).

**Comment by peterjohnlawrence** [comment]
I've used Halyard and was very impressed. The source is readily available and surprisingly simple. It does need to be updated to the latest RDF4J version, but I would suggest it is well worth the effort.
